### DISCUSSION TITLE
Are there any "limited mode" or "secure mode" of qalc?

### DISCUSSION THREAD

**Discussion by hv0905** [Qalculate/libqalculate#495] [Q&A] [answered]
I want to expose my `qalc` CLI interface to a web API or a chatbot, however, I noticed that there are some functions such as `command` can execute any command on my remote server and lead to security problem.

So I'm hoping to have a `limited mode` for `qalc` which is secure to be exposed to untrusted environment. If the `limited mode` is unavailable, I want to have a full list of which function or keywork I should ban on by server backend.

Any help will be thanked!

**Comment by hanna-kn** [comment, marked as answer]
There is a compile-time configuration option `--without-gnuplot-call` which disables external commands (`command` and `plot` functions).

**Reply by hv0905** [reply]
Thanks! I will try it out
